app/models/blog.rb

```diff
@@ -10,11 +10,19 @@ class Blog < ApplicationRecord
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where("title LIKE '%#{term}%' OR content LIKE '%#{term}%'")
+    where('title LIKE :term OR content LIKE :term', term: "%#{term}%")
   }
 
   scope :default_order, -> { order(id: :desc) }
 
+  scope :view_auth, lambda { |user|
+    if user
+      where('secret = FALSE').or(where(user_id: user.id))
```

Comment:
```suggestion
      where(secret: false).or(where(user_id: user.id))
```

って書いた方がRailsらしいですね。
ちなみに10行目は、「Railsがよくわかってない人が書いたコード（だからSQLインジェクションとかを作り込んじゃう）」をシミュレートしたものです 😅

Reply:
そもそも`where(secret: false)`はpublishedのscopeを再利用できます

Reply:
> 「Railsがよくわかってない人が書いたコード

なるほど、、そんな意図があったんですね、
[refactor: 不要な条件分岐削除](https://github.com/sho037/haunted-blog/pull/1/commits/3e97dd10ae8f1196c7a425ccd553a886d8e18ddc) こちらで変更しました。